src/ORM/DataList.php

```diff
@@ -1398,6 +1421,33 @@ private function addEagerLoadedDataToParent(
         }
     }
 
+    /**
+     * NOTE: Do not change `DataList` to `static` in this method signature.
+     * Subclasses of DataList must still accept DataList arguments and return DataList!
+     */
+    private function manipulateEagerLoadingQuery(
+        DataList $fetchList,
+        string $relationChain,
+        string $relationType
+    ): DataList {
```

Comment:
Discussed with Steve - he prefers `DataList` here over `self`.
I'm agnostic - I think the comment needs to be there for our future selves either way, and both IDEs and static analysis will ultimately treat them the same.